app/Http/Controllers/MetricsController.php

```diff
@@ -21,6 +27,19 @@
  */
 class MetricsController extends Controller
 {
+	public function get(MetricsRequest $request, GetMetrics $get_metrics, CleanupMetrics $cleanup_metrics): Collection
+	{
+		if (Configs::getValueAsBool('live_metrics_enabled') === false) {
+			throw new UnauthorizedException('Live metrics are not enabled.');
+		}
+
+		// First clean up.
+		$cleanup_metrics->do();
```

Comment:
Is there a risk of this delete query negatively impacting user experience, and would it be worth moving the cleanup to a scheduled command or job so it runs in the background?

Reply:
The clean up is just a DELETE on the db with anything older than XXX.
That endpoint is also only queried on demand if the user has access rather than for everyone.
So I am not sure whether it would impact negatively or not.

I do agree that having a scheduled command/job could also do the trick (or do a `defer()` call).